File: Backend/UserDataService/src/main/java/com/user/dto/LeaderBoardOnFootprint.java
package com.user.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class LeaderBoardOnFootprint {
	private String name;
	private String city;
	private Double totalFootprint;
	private boolean isCurrentUser = false;
	
	public LeaderBoardOnFootprint(String name, String city, Double totalFootprint) {
		this.name = name;
		this.city = city;
		this.totalFootprint = totalFootprint;
	}
}

File: FrontEnd/Self-Tracker/src/app/models/leader-board-on-footprint.ts
export class LeaderBoardOnFootprint {
  name: string;
  city: string;
  totalFootprint: number;
  isCurrentUser: boolean;

  constructor(name: string, city: string, totalFootprint: number, isCurrentUser: boolean) {
    this.name = name;
    this.city = city;
    this.totalFootprint = totalFootprint;
    this.isCurrentUser = isCurrentUser;
  }
}
